http_client.rst

```diff
@@ -631,6 +631,15 @@ according to the ``multipart/form-data`` content-type. The
         'body' => $formData->bodyToIterable(),
     ]);
 
+HttpClient will stream the upload of the body by default. This might not work with all
+servers, resulting in HTTP status code 411 ("Length Required") cause there is no
+``Content-Length`` header. In this case, just turn the body into a string::
```

Comment:
```suggestion
HttpClient will stream the upload of the body by default. Some clients, e.g. curl, are not able to send a ``Content-Length`` header when you stream the upload. This can lead to servers rejecting the request with a HTTP status code 411 ("Length Required"). If you encounter this problem, you can turn the body into a string instead::
```

Reply:
> Some clients, e.g. curl, are not able to send a ``Content-Length`` header when you stream the upload.

I'm certainly no expert here! But as far I understand/guess it, you can never provide a `Content-Lengh` if you stream. And curl streams by default. Right?

Reply:
As I understood it, in the linked issue, NativeClient has a way of calculating the header. So it is a matter of which implementation of HttpClient you use. I might be wrong, though.

Reply:
See https://github.com/symfony/symfony/issues/34951#issuecomment-567489789

> Because NativeHttpClient cannot stream the upload, it has to turn the body into a string before doing the request, thus it can compute the content-Length.

As I understand it, it's not about the client, but about the method: If you stream, you cannot have a `Content-Length`; if you upload as string, you can. @nicolas-grekas: What's right here?

Reply:
Both curl and native clients send `Content-Length` when the body is a string.
Only curl knows how to stream uploads (native casts the stream to a string, which consumes memory for large streams).
So this is not implementation dependent: whether curl or native, if you must send `Content-Length`, you must cast to string before.

Reply:
(or compute the `Content-Length` on your own of course)